platform/src/main/java/org/hillview/sketches/Centroids.java

```diff
@@ -0,0 +1,101 @@
+package org.hillview.sketches;
+
+import org.hillview.dataset.api.IJson;
+import org.hillview.table.api.IMembershipSet;
+import org.hillview.table.api.ITable;
+import org.hillview.utils.BlasConversions;
+import org.jblas.DoubleMatrix;
+import org.jblas.ranges.AllRange;
+import org.jblas.ranges.PointRange;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * This class represents a set of centroids of a partitioning of a table. The number of centroids should not be too
+ * large, as they are shipped over the network.
+ */
+public class Centroids implements IJson {
+    /**
+     * Encodes the position of the centroids. Every row is a centroid, and the columns are the dimensions.
+     * Each partition has a centroid, which is simply the mean of the nD points in the partition.
+     * Should be computed with computeCentroids() by the head node.
+     */
+    @Nullable
+    public DoubleMatrix centroids;
+    /**
+     * Holds the sum of the values in the partitions. Every row is the sum of all rows belonging to the corresponding
+     * partition.
+     */
+    public DoubleMatrix sums;
+    /**
+     * Registers how many rows are represented by every centroid.
+     */
+    public long[] counts;
+
+    /**
+     * Constructs a zero-centroids. Needs to know the dimensions.
+     * @param numPartitions Number of partitions.
+     * @param numColumns Number of columns (= number of dimensions in the centroid space).
+     */
+    public Centroids(int numPartitions, int numColumns) {
```

Comment:
Another benefit is that you don't need to know the number of partitions anymore; any key that does not exist in the hashmap implies a zero doublevector.